spyder/plugins/ipythonconsole/widgets/shell.py

```diff
@@ -873,9 +918,15 @@ def insert_horizontal_ruler(self):
         self._control.insert_horizontal_ruler()
 
     # ---- Public methods (overrode by us) ------------------------------------
-    def request_restart_kernel(self):
-        """Reimplemented to call our own restart mechanism."""
-        self.ipyclient.restart_kernel()
+    def _event_filter_console_keypress(self, event):
```

Comment:
I think this new method overrides the `_event_filter_console_keypress` one in DebuggingWidget (given that that class is sort of a mixin). If that's the case, this code should be moved there.

Reply:
Unless I am mistaken, `return super()._event_filter_console_keypress(event)` will call `DebuggingWidget._event_filter_console_keypress`.
I can confirm I see that when debugging Spyder on my mac.

Reply:
You're right, sorry for the noise. I just wasn't sure about this.